google-cloud-spanner/src/main/java/com/google/cloud/spanner/ReplicaInfo.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spanner;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.Objects;
+
+/** Represents a Cloud Spanner replica information. */
+public class ReplicaInfo {
+
+  abstract static class Builder {
+    abstract Builder setLocation(String location);
+
+    abstract Builder setType(ReplicaType type);
+
+    abstract Builder setDefaultLeaderLocation(boolean defaultLeaderLocation);
+
+    abstract Builder setProto(com.google.spanner.admin.instance.v1.ReplicaInfo proto);
+
+    public abstract ReplicaInfo build();
+  }
+
+  public static class BuilderImpl extends Builder {
+
+    private String location;
+    private ReplicaType type;
+    private boolean defaultLeaderLocation;
+    private com.google.spanner.admin.instance.v1.ReplicaInfo proto;
+
+    /** The location of the serving resources. */
```

Comment:
Would it be possible (and make sense) to reference some online documentation that indicates what the possible values are, including the format that is expected?

Reply:
Updated the javadoc with the link to the possible values here.